sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -1062,4 +1062,11 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
       checkAnswer(dfNonBinary, dfBinary)
     }
   }
+
+  test("SPARK-47765: test SET COLLATION statement") {
```

Comment:
I'm looking at how we test `SET TIMEZONE`. It's in `SQLConfSuite`. I feel it's a better way of testing as we are making tests orthogonal. We need to make sure `SET COLLATION` sets the expected conf correctly, and we already have test in this suite to make sure the default collation config works.

Reply:
Moved the test and added a check for default collation name, in order to throw error immediately when invalid name is given.